friends/src/App.js

```diff
@@ -35,17 +31,42 @@ class App extends Component {
     })
   }
   
+  handleSubmit = newFriend =>{
+    axios.post('http://localhost:5000/friends', newFriend)
+   .then(res=>{
+     this.setState({friends: res.data})
+   })         
+   .catch(err=>{console.log('Error: ', err)});
+}
+
+delFriend = id =>{
+  console.log('water!')
+  axios.delete(`http://localhost:5000/friends/${id}`)
+   .then(res=>{
+     this.setState({friends: res.data})
+   })         
+   .catch(err=>{console.log('Error: ', err)});
+}
+
+updateFriend = friend => {
+  console.log('fire!')
+  this.setState({currentFriend:friend});
+  this.props.history.push('/');
```

Comment:
The App component does not seem to have access to the history prop. If you would like to give it access, you can use `withRouter`